testsuite/MDAnalysisTests/core/test_accumulate.py

```diff
@@ -0,0 +1,165 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from __future__ import division, absolute_import
+
+import numpy as np
+from numpy.testing import assert_equal, assert_almost_equal
+
+import MDAnalysis as mda
+from MDAnalysis.exceptions import DuplicateWarning, NoDataError
+from MDAnalysisTests.datafiles import PSF, DCD, TPR_xvf, TRR_xvf, GRO
+import pytest
+
+universe = mda.Universe(PSF, DCD)
+universe_molfrag = mda.Universe(TPR_xvf, TRR_xvf)
+levellist = ('atoms', 'residues', 'segments')
+
+
+class TestAccumulate(object):
+    """Tests the functionality of *Group.accumulate()."""
+
+    @pytest.mark.parametrize('level', levellist)
+    def test_accumulate_str_attribute(self, level):
```

Comment:
so here you could do something like:

```python
@pytest.fixture(params['atoms', 'residues' 'segments'])
def group(request):
    u = mda.Universe(..., ...)
    return getattr(u, request.params[0])

def test_accumulate_str_attribute(group):
    assert_almost_equal(group.accumulate(), np.sum())
```
and the fixture params make the test expand to cover all 3 types of Group